python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -312,10 +312,6 @@ def checkExistingOutputs(self, quantum: Quantum, taskDef: TaskDef, limited_butle
         RuntimeError
             Raised if some outputs exist and some not.
         """
-        if not self.butler:
-            # Skip/prune logic only works for full butler.
-            return False
```

Comment:
@andy-slac, do you recall why we made this shortcut?  At first I thought it must be because `LimitedButler` didn't have `pruneDatasets`...but it does.  So this is either something we should have done a while ago (but maybe thought we didn't need) or there's something deeper.

This is all coming up because @mfisherlevine now has a `LimitedButler` that wraps a another `LimitedButler` (which is actually a full one in this use case) and does in-memory caching, and he's got a use case for that to be able to clobber.

Reply:
This is all passing, but I'll wait to hear back from @andy-slac before merging anything.

Reply:
I _think_ this was because back then we needed access to `butler.datastore` for implementation of this method and limited butler did not provide that property. We had extended limited butler interface since then, so it should be OK now to allow limited butler here.